### PR TITLE
[java] Update rule SwitchDensity

### PR DESCRIPTION
**Pull request by oowekyala**
## Describe the PR

This also adds support for switch expressions, composed switch labels (eg `case 0, 1, 2:`) and switch arrow branches for free 

## Related issues

- Refs #2701

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to pmd/7.0.x:<br />
This changeset changes 0 violations,<br />
introduces 77 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/2d07dc03c6c349b08c6cceb6b43be2aa/diff1/index.html">Full report</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 31204 violations,<br />
introduces 21184 new violations, 3 new errors and 0 new configuration errors,<br />
removes 137925 violations, 8 errors and 3 configuration errors.<br />
<a href="https://chunk.io/pmd/2d07dc03c6c349b08c6cceb6b43be2aa/diff2/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SwitchDensityRule.java [hunk 1]
```diff
@@ -36,51 +37,32 @@
...
-            return data;
-        }
-
-        @Override
-        public Object visit(ASTSwitchLabel switchLabel, Object data) {
-            if (switchLabel.getParent() == root) {
-                labels++;
-            }
-            return super.visit(switchLabel, data);
+        // note: if labelCount is zero, double division will produce NaN, not ArithmeticException
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SwitchDensityRule.java:61]: hm... division by zero seems to produce (positive) infinity (not NaN) - which is always bigger than the report level... hence the FP with only a default label...

Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SwitchDensityRule.java:61]:
```suggestion
        // note: if labelCount is zero, double division will produce +Infinity or NaN, not ArithmeticException
```
+Infinity if stmtCount > 0 and labelCount is zero, NaN if both stmtCount and labelCount is zero

Comment (oowekyala) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SwitchDensityRule.java:61]: Ah that's right, thanks! I had only checked `0/0.`

**Pull request merged**